### PR TITLE
Fixed tests

### PR DESCRIPTION
**Pull request by gpitton**
All tests are working with python 3

### CODE REVIEW

**Comment by luca-heltai** [comment]
I think you need to `apt-get install gfortran` before you can install igakit

**Comment by gpitton** [comment]
Indeed, thanks.

File: interfaces/bspline_vector_space.py [hunk 1]
```diff
@@ -13,11 +13,10 @@ class BsplineVectorSpace(VectorSpace):
     def __init__(self, degree=0, knots=[]):
         """ Pure interface class. It generates the constant on [0,1] if no arguments are provided. """
         assert degree >= 0
-        assert len(knots) > 1
         assert knots[0] != knots[-1]
```
Comment (luca-heltai) [inline interfaces/bspline_vector_space.py]: move this below, otherwise you'll get an assertion when trying to access an empty knot list

**Comment by luca-heltai** [comment]
I think that `self.knots` should be an array, and not a list. At the end of the constructor in BsplineVectorSpace, add a `self.knots = np.array(self.knots)`, to get rid of all `typerror` in the tests.

The only remaining failing test, is the one of the fourth derivative. You can safely change that test to only test up to derivatives of order 3 (for the moment).

**Comment by gpitton** [comment]
Currently I fixed this error by using `np.asarray(self.knots) ` in the comparison. This does not create a copy of self.knots, so there should not be much overhead. Is it better to change the definition of `self.knots`, and make in a `np.array`?

**Comment by luca-heltai** [comment]
No, this is ok. There is still a problem with the fourth order derivative.

**Pull request merged**